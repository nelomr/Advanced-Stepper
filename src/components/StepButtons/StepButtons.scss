.step-buttons {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    button + button {
        margin-left: 0;
    }
}

a.step-button {
    justify-content: center;
    text-decoration: none;
    transition: all .3s ease;

    &.has-margin-top {
        margin-top: 40px;
    }

    &:hover {
        color: $light-color;
        background-color: $line-color;
    }
}

.step-button {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    min-width: 170px;
    padding: 8px 10px;
    margin-bottom: 10px;
    outline: none;
    appearance: none;
    font-weight: bold;
    font-size: 18px;
    line-height: 25px;
    color: $line-color;
    cursor: pointer;
    background-color: $light-color;
    border: 2px solid $line-color;
    box-sizing: border-box;
    border-radius: 4px;

    @include breakpoint($min-tablet) {
        width: auto;
        margin-bottom: 0;
    }

    i {
        display: flex;
        align-items: center;
        justify-content: center;
    }

    &.is-next {
        color: $light-color;
        background-color: $line-color;
    }

    &.is-hidden {
        opacity: 0;
        visibility: hidden;
    }
}
