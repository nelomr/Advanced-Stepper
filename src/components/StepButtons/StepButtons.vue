<template>
  <div class="step-buttons">
    <button 
      @click="prev()"
      :class="{'is-hidden': currentStep < 1}" 
      class="step-button is-prev"
    >
      <i><IconButtonLeft></IconButtonLeft></i>
      Anterior
    </button>
    <button
      v-if="currentStep < 4"
      @click="next()"
      class="step-button is-next"
    >
      Continuar
      <i><IconButtonRight></IconButtonRight></i>
    </button>
  </div>
</template>

<script>
import IconButtonRight from '@/components/icons/IconButtonRight.vue';
import IconButtonLeft from '@/components/icons/IconButtonLeft.vue';

export default {
  components: {
    IconButtonRight,
    IconButtonLeft
  },
  computed: {
    currentStep() {
        return this.$store.state.step;
    }
  },
  methods: {
    next() {
      if (this.currentStep < 4) {
        this.$store.dispatch('incrementStep');
      }
    },
    prev() {
      if (this.currentStep > 0) {
        this.$store.dispatch('decreaseStep');
      }
    }
  }
}
</script>

<style lang="scss">
    @import "./StepButtons.scss";
</style>