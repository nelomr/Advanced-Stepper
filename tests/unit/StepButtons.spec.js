import Vue from 'vue';
import Vuex from 'vuex';
import {shallowMount, createLocalVue} from '@vue/test-utils';
import StepButtons from '@/components/StepButtons/StepButtons';
import IconButtonRight from '@/components/icons/IconButtonRight.vue';
import IconButtonLeft from '@/components/icons/IconButtonLeft.vue';
Vue.component('IconButtonRight', IconButtonRight);
Vue.component('IconButtonLeft', IconButtonLeft);

const localVue = createLocalVue();
localVue.use(Vuex);

describe('StepButtons', () => {
    let step = 0;
    let store;
    let state;
    let actions;

    beforeEach(() => {
        state = {
            step
        };

        actions = {
            incrementStep: jest.fn(),
            decreaseStep: jest.fn()
        };

        store = new Vuex.Store({
            state,
            actions
        });
    });

    const build = () => {
        const wrapper = shallowMount(StepButtons, {
            mocks: {$store: store},
            computed: {
                currentStep() {
                    return store.state.step;
                }
            }
        });

        return {
            wrapper
        };
    };

    it('renders the component', () => {
        const {wrapper} = build();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('calls store action "incrementStep" when button is clicked', () => {
        const build = () => {
            const wrapper = shallowMount(StepButtons, {
                mocks: {$store: store},
                computed: {
                    currentStep() {
                        return store.state.step;
                    }
                }
            });

            return {
                wrapper,
                button: () => wrapper.find('.step-button.is-next')
            };
        };

        const {button} = build();

        button().trigger('click');
        expect(actions.incrementStep).toHaveBeenCalled();
    });

    it('with step < 1 button prev is hidden class', () => {
        const build = () => {
            const wrapper = shallowMount(StepButtons, {
                mocks: {$store: store},
                computed: {
                    currentStep() {
                        return store.state.step;
                    }
                }
            });

            return {
                wrapper,
                button: () => wrapper.find('.step-button.is-prev')
            };
        };

        const { button} = build();

        expect(button().classes('is-hidden')).toBe(true);
    });

    it('calls store action "decreaseStep" when button is clicked', () => {
        state = {
            step: 2
        };

        actions = {
            incrementStep: jest.fn(),
            decreaseStep: jest.fn()
        };

        store = new Vuex.Store({
            state,
            actions
        });
        const build = () => {
            const wrapper = shallowMount(StepButtons, {
                mocks: {$store: store},
                computed: {
                    currentStep() {
                        return store.state.step;
                    }
                }
            });

            return {
                wrapper,
                button: () => wrapper.find('.step-button.is-prev')
            };
        };

        const { button} = build();

        button().trigger('click');
        expect(actions.decreaseStep).toHaveBeenCalled();
    });
});
