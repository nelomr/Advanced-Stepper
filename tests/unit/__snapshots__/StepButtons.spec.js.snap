// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepButtons renders the component 1`] = `
<div class="step-buttons"><button class="step-button is-prev is-hidden"><i>
      <iconbuttonleft-stub></iconbuttonleft-stub>
    </i>
    Anterior
  </button> <button class="step-button is-next">
    Continuar
    <i>
      <iconbuttonright-stub></iconbuttonright-stub>
    </i></button></div>
`;
